hazelcast-jet-core/src/main/java/com/hazelcast/jet/Jet.java

```diff
@@ -125,33 +126,32 @@ static void configureJetService(JetConfig jetConfig) {
 
         HazelcastProperties properties = new HazelcastProperties(hzProperties);
 
-        ServicesConfig servicesConfig = hzConfig.getServicesConfig();
-        servicesConfig
-                .addServiceConfig(new ServiceConfig().setEnabled(true)
+        hzConfig.getServicesConfig()
+                .addServiceConfig(new ServiceConfig()
+                        .setEnabled(true)
                         .setName(JetService.SERVICE_NAME)
                         .setClassName(JetService.class.getName())
                         .setProperties(jetServiceProperties(properties))
-                        .setConfigObject(jetConfig));
-
-        servicesConfig
-                .addServiceConfig(new ServiceConfig().setEnabled(true)
+                        .setConfigObject(jetConfig))
+                .addServiceConfig(new ServiceConfig()
+                        .setEnabled(true)
                         .setName(JetMetricsService.SERVICE_NAME)
                         .setClassName(JetMetricsService.class.getName())
                         .setConfigObject(jetConfig.getMetricsConfig()));
 
-
-        MapConfig metadataMapConfig = new MapConfig(INTERNAL_JET_OBJECTS_PREFIX + "*")
+        MapConfig metadataMapConfig = new MapConfig(INTERNAL_JET_OBJECTS_PREFIX + '*')
                 .setBackupCount(jetConfig.getInstanceConfig().getBackupCount())
                 .setStatisticsEnabled(false)
                 .setMergePolicyConfig(
-                        new MergePolicyConfig().setPolicy(IgnoreMergingEntryMapMergePolicy.class.getName()));
+                        new MergePolicyConfig().setPolicy(IgnoreMergingEntryMapMergePolicy.class.getName()))
+                .setHotRestartConfig(new HotRestartConfig().setEnabled(true));
 
         MapConfig resultsMapConfig = new MapConfig(metadataMapConfig)
                 .setName(JOB_RESULTS_MAP_NAME)
                 .setTimeToLiveSeconds(properties.getSeconds(JOB_RESULTS_TTL_SECONDS));
+        resultsMapConfig.getHotRestartConfig().setEnabled(true);
```

Comment:
What if the user wants to enable hot restart, but not for jet jobs? We unconditionally enable it?